### ISSUE TITLE
Gitcoin Contribution not updated on the Grants page

### ISSUE THREAD

**Issue by arcology0** [gitcoinco/web#10322]
Hello there,

As discussed in the discord channel 'funder-540', I donated 0.31 ETH to the ManagerDAO grant, and sped up the transaction through my Metamask. However, the grants page did not update while my transaction went through successfully. 

ManagerDAO page: https://gitcoin.co/grants/4962/managerdao-democratising-sports-team-ownership

**Steps To Reproduce**
1. Grant added to cart.
2. Checkout using standard method.
3. The transaction tab went up, and corresponding signature was given on metamask.
4. I sped up the transaction.
5. The metamask transaction was successfully executed, but the gitcoin transaction tab did not update.
6. Grant amount did not update.

**Observed behavior**
Gitcoin grant amount was not recorded, but metamask transaction (contract execution) was successful.

**Expected behavior**
Gitcoin grant amount recorded and my activity would be updated.

**Transaction TX**
https://etherscan.io/tx/0x5c416369d9d1347ff14cd0e3b1e9b3f0ffabf03f258abb15df4745d252a2cd06

**Additional Notes**
As recommended by user @Sirlupinwatson on discord help ticket channel, this appears to be similar to issue #7298.

Appreciate any help to get the transaction updated or the contribution amount refunded so I can try again. Thank you in advance!

**Comment by nfrgosselin** [comment]
Hey @arcology0 — thanks for reporting this! Sorry that you experienced this issue. We have a backend script that should pull in missing contributions from the blockchain, so I'm not entirely sure why this hasn't been updated yet. I'm going to pull this issue into the Grants team's backlog as a bug. Stay tuned!

**Comment by thelostone-mc** [comment]
Ah the speed up on transaction is likely why it didn't get picked up cause we rely on etherscan API which doesn't show what the replaced txnhash via the API. 
Meanwhile -> you could add your contribution via https://gitcoin.co/grants/add-missing-contributions

**Comment by sirlupinwatson** [comment]
> Ah the speed up on transaction is likely why it didn't get picked up cause we rely on etherscan API which doesn't show what the replaced txnhash via the API. Meanwhile -> you could add your contribution via https://gitcoin.co/grants/add-missing-contributions

There might be another issue, in my case adding contribution using this link does not work.
"Error adding contributions as Profile could not be found"

I suppose this can be in relation with changing my GitHub username a few months ago (I know this is another issue) or not but there is still another glitch if you go on Townsquare and put your mouse over your name, I can't see anymore my contribution in the small rectangle (Tip send, Receive) (It says 1 Tip send but actually I did send over 175 Tip I think)

Just a small () Hope this can help! Thanks @thelostone-mc

**Comment by arcology0** [comment]
@thelostone-mc I also came across an issue when using the form.

I inputted the transaction hash '0x5c416369d9d1347ff14cd0e3b1e9b3f0ffabf03f258abb15df4745d252a2cd06' while connected to ETH mainnet and an error came up 'ethereum is not defined'. Do you know what this means?

**Comment by thelostone-mc** [comment]
@sirlupinwatson the profile change is a diff issue and I believe you'd have to raised a ticket against customer support and they'd help merge the two accounts into 1.

**Comment by thelostone-mc** [comment]
> I inputted the transaction hash '0x5c416369d9d1347ff14cd0e3b1e9b3f0ffabf03f258abb15df4745d252a2cd06' while connected to ETH mainnet and an error came up 'ethereum is not defined'. Do you know what this means?

Hmm my hunch is that web3 wasn't injected by metamask. So this means either you didn't metamask installed / you have a plugin that prevents metamask from injecting it's web3 instance
This is needed cause while ingesting -> you will be asked to sign a message to prove ownership of the account


> However, the Grants page have not been updated with my contribution. Is this an instantaneous process?

It is an instantaneous process. The reason is doesn't show up is cause we don't have a grant mapped to that address in our database. This could be cause grant owner closed their grant / changed their grant address.
So when attempting to ingest by address -> we can't figure out where to map the txn to 

in https://etherscan.io/tx/0x5c416369d9d1347ff14cd0e3b1e9b3f0ffabf03f258abb15df4745d252a2cd06
the funds were sent to https://etherscan.io/address/0x7adaf30ebeda34058f8ee221e9171267f2cc7dce
and looking at the history it had 7 txns which leads me to belive the grant owner changed the address at gitcoin which is why we cannot map it


<img width="1115" alt="image" src="https://user-images.githubusercontent.com/5358146/159211703-d5e56e28-094e-4b65-84ab-02a0d954dbb2.png">

**Comment by thelostone-mc** [comment]
@nfrgosselin the improvements we could is updating the error message from 'ethereum is not defined to say something like metmask not installed / locked / etc 
I'm not sure how to reproduce this so once we get those steps -> we could throw a nicer error message for that

**Comment by arcology0** [comment]
> > I inputted the transaction hash '0x5c416369d9d1347ff14cd0e3b1e9b3f0ffabf03f258abb15df4745d252a2cd06' while connected to ETH mainnet and an error came up 'ethereum is not defined'. Do you know what this means?
> 
> Hmm my hunch is that web3 wasn't injected by metamask. So this means either you didn't metamask installed / you have a plugin that prevents metamask from injecting it's web3 instance This is needed cause while ingesting -> you will be asked to sign a message to prove ownership of the account
> 
> =========
> 
> > However, the Grants page have not been updated with my contribution. Is this an instantaneous process?
> 
> It is an instantaneous process. The reason is doesn't show up is cause we don't have a grant mapped to that address in our database. This could be cause grant owner closed their grant / changed their grant address. So when attempting to ingest by address -> we can't figure out where to map the txn to
> 
> in https://etherscan.io/tx/0x5c416369d9d1347ff14cd0e3b1e9b3f0ffabf03f258abb15df4745d252a2cd06 the funds were sent to https://etherscan.io/address/0x7adaf30ebeda34058f8ee221e9171267f2cc7dce and looking at the history it had 7 txns which leads me to belive the grant owner changed the address at gitcoin which is why we cannot map it
> 
> <img alt="image" width="1115" src="https://user-images.githubusercontent.com/5358146/159211703-d5e56e28-094e-4b65-84ab-02a0d954dbb2.png">

noted regarding the first error - I did not have metamask extension installed on one of my computers so filled in the form with my mobile device.

However, for the second issue about the instantaneous update...this needs a bit more investigation. Adding @bberry259 into the conversation. have you changed the grant address per @thelostone-mc 's mention above?

thank you all!

**Comment by bberry259** [comment]
Hi Team.

We did indeed change the address to an EOA so that we could accept Polygon.

That said I can still see all other donations made prior on Gitcoin. Both
donations from Arcology are showing up on chain. Just one isn't showing up
on gitcoin. I understand both were made through the UI checkout process.

**Comment by nfrgosselin** [comment]
> @nfrgosselin the improvements we could is updating the error message from 'ethereum is not defined to say something like metmask not installed / locked / etc I'm not sure how to reproduce this so once we get those steps -> we could throw a nicer error message for that

Makes sense to me — the story for the change is [here](https://github.com/gitcoinco/web/issues/10383) FYI

**Comment by arcology0** [comment]
@thelostone-mc @nfrgosselin - I have discussed with @bberry259 and we have agreed that a workaround for this is to have the project 'refund' the ETH I have donated and I will attempt to donate again. As the initial donation amount has not been picked up by gitcoin, I believe that this will be easiest way forward to re-register the donation amounts with gitcoin. 

Do let us know whether this is an acceptable course of action short of gitcoin manually registering the transaction. Thanks!
